dbms/src/Storages/DeltaMerge/Index/MinMaxIndex.cpp

```diff
@@ -328,68 +252,38 @@ RSResults MinMaxIndex::checkNullableIn(
-        for (size_t i = start_pack; i < start_pack + pack_count; ++i)                          \
-        {                                                                                      \
-            bool min_is_null = null_map.getElement(i * 2);                                     \
-            if (min_is_null)                                                                   \
-                continue;                                                                      \
-            auto min = minmaxes_data[i * 2];                                                   \
-            auto max = minmaxes_data[i * 2 + 1];                                               \
-            results[i - start_pack] = RoughCheck::checkIn<TYPE>(values, type, min, max);       \
-        }                                                                                      \
-        return results;                                                                        \
-    }
+#define DISPATCH(TYPE)                                 \
+    if (typeid_cast<const DataType##TYPE *>(raw_type)) \
+        return checkNullableInImpl<TYPE>(column_nullable, null_map, start_pack, pack_count, values, type);
     FOR_NUMERIC_TYPES(DISPATCH)
 #undef DISPATCH
     if (typeid_cast<const DataTypeDate *>(raw_type))
-    {
-        const auto & minmaxes_data = toColumnVectorData<DataTypeDate::FieldType>(column_nullable.getNestedColumnPtr());
-        for (size_t i = start_pack; i < start_pack + pack_count; ++i)
-        {
-            bool min_is_null = null_map.getElement(i * 2);
-            if (min_is_null)
-                continue;
-            auto min = minmaxes_data[i * 2];
-            auto max = minmaxes_data[i * 2 + 1];
-            results[i - start_pack] = RoughCheck::checkIn<DataTypeDate::FieldType>(values, type, min, max);
-        }
-        return results;
-    }
+        return checkNullableInImpl<DataTypeDate::FieldType>(
+            column_nullable,
+            null_map,
+            start_pack,
+            pack_count,
+            values,
+            type);
     if (typeid_cast<const DataTypeDateTime *>(raw_type))
-    {
-        const auto & minmaxes_data
-            = toColumnVectorData<DataTypeDateTime::FieldType>(column_nullable.getNestedColumnPtr());
-        for (size_t i = start_pack; i < start_pack + pack_count; ++i)
-        {
-            bool min_is_null = null_map.getElement(i * 2);
-            if (min_is_null)
-                continue;
-            auto min = minmaxes_data[i * 2];
-            auto max = minmaxes_data[i * 2 + 1];
-            results[i - start_pack] = RoughCheck::checkIn<DataTypeDateTime::FieldType>(values, type, min, max);
-        }
-        return results;
-    }
+        return checkNullableInImpl<DataTypeDateTime::FieldType>(
+            column_nullable,
+            null_map,
+            start_pack,
+            pack_count,
+            values,
+            type);
```

Comment:
We'd better move `DataTypeDate` and `DataTypeDateTime` to the last of this function. Because actually they are deadcode as we use `DataTypeMyDateTime`/`DataTypeMyDate` in tidb.